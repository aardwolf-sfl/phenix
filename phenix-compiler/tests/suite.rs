use std::path::Path;

use phenix_codegen::Project;
use phenix_compiler::Compiler;

fn compile<P: AsRef<Path>>(path: P) -> Project {
    Compiler::new().compile(path.as_ref().try_into().unwrap())
}

#[test]
fn basic() {
    let project = compile("tests/schemas/basic/index.phenix");
    insta::assert_yaml_snapshot!(project);
}

#[test]
fn imports() {
    let project = compile("tests/schemas/imports/index.phenix");
    insta::assert_yaml_snapshot!(project);
}
