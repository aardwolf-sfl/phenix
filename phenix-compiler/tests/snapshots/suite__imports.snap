---
source: phenix-compiler/tests/suite.rs
expression: project
---
modules:
  - id: 0
    path: []
    types: []
  - id: 1
    path:
      - a
    types:
      - type: Struct
        id: 0
        name: A
        fields:
          - name: shared
            ty:
              id:
                user: 2
              generics: []
            attrs: []
        attrs: []
  - id: 2
    path:
      - c
    types:
      - type: Struct
        id: 2
        name: Shared
        fields:
          - name: pair
            ty:
              id:
                user: 1
              generics: []
            attrs: []
        attrs: []
  - id: 3
    path:
      - nested
      - x
    types:
      - type: Struct
        id: 1
        name: PairOfNumbers
        fields:
          - name: first
            ty:
              id:
                builtin: uint
              generics: []
            attrs: []
          - name: second
            ty:
              id:
                builtin: uint
              generics: []
            attrs: []
        attrs: []
  - id: 4
    path:
      - b
    types:
      - type: Struct
        id: 3
        name: B
        fields:
          - name: shared
            ty:
              id:
                user: 2
              generics: []
            attrs: []
        attrs: []

